src/core/modules/impl/HandlerExecutorFactory.ts

```diff
@@ -266,6 +267,8 @@ export interface HandlerApi<State> {
   initState(state: State): void;
 
   maybeCallStateConstructor(initialState: State, executionContext: ExecutionContext<State>): Promise<State>;
+
+  evolveState(): Promise<EvolveState>;
```

Comment:
Why is it here? Shouldn't it rather be in HandlerApi interface?

Reply:
wow. This shouldn't be here. I'll remove it.